core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,332 @@
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = change_cmd
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type
+        SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+
+        Raises:
+            ValidationError: One or more attributes of the
+                SuggestionEditStateContent object are invalid.
+        """
+        super(SuggestionEditStateContent, self).validate()
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_CONTENT:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
+                ', received %s' % (
+                    suggestion_models.SCORE_TYPE_CONTENT,
+                    self.get_score_type()))
+
+        if self.get_score_sub_type() not in constants.ALL_CATEGORIES:
+            raise utils.ValidationError(
+                'Expected the second part of score_category to be a valid'
+                ' category, received %s' % self.get_score_sub_type())
+
+        if 'cmd' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a cmd key')
+
+        if self.change_cmd['cmd'] != exp_domain.CMD_EDIT_STATE_PROPERTY:
+            raise utils.ValidationError(
+                'Expected cmd to be %s, received %s' % (
+                    exp_domain.CMD_EDIT_STATE_PROPERTY, self.change_cmd['cmd']))
+
+        if 'property_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a property_name key')
+
+        if (
+                self.change_cmd['property_name'] !=
+                exp_domain.STATE_PROPERTY_CONTENT):
+            raise utils.ValidationError(
+                'Expected property_name to be %s, received %s' % (
+                    exp_domain.STATE_PROPERTY_CONTENT,
+                    self.change_cmd['property_name']))
+
+        if 'state_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a state_name key')
+
+        states = exp_services.get_exploration_by_id(self.target_id).states
+        if self.change_cmd['state_name'] not in states:
+            raise utils.ValidationError(
+                'Expected %s to be a valid state name' %
+                self.change_cmd['state_name'])
+
+    def update_change_cmd_before_accept(self):
+        """Before accepting the suggestion, modifications need to be done to the
+        change_cmd.
+        """
+        exploration = exp_services.get_exploration_by_id(self.target_id)
+        old_content = exploration.states[self.change_cmd['state_name']].content
+
+        self.change_cmd['old_content'] = old_content
+
+        self.change_cmd['new_content']['audio_translations'] = (
+            old_content['audio_translations'])
+
+    def accept(self, commit_message):
```

Comment:
Is this tested anywhere?

Reply:
This function is actually a tricky one to unit test. I thought about it for a while and I'm not sure how to test this. I decided that when we write a final integration test for the framework, we can test the behavior effectively. 

So to unit test the issue is we have to create an exploration, add a state, and create a valid suggestion, and then accept suggestion. Once accepted, validate new content is the same as the suggested content -- This feels more like an integration test rather than a unit test. Also the accept_suggestion (service) which calls this method is tested by mocking update_exploration.

What do you suggest should be done here?

Reply:
Yeah, I would expect an integration test (without mocking update_exploration). Is there a reason we can't do that now?

Reply:
Nothing blocking this. I thought we'll add the integration test at the controller level (by sending requests to the appropriate URLs). But I think it makes sense to have an integration here too. Will add one.

Reply:
Still working on this. Will push this soon.